src/ngx_http_lua_ssl_module.c

```diff
@@ -303,9 +322,15 @@ ngx_http_lua_ffi_update_last_ticket_decryption_key(SSL_CTX *ctx,
     pkey = &pkey[keys->nelts-1];
 
     dd("replace the last key");
-    ngx_memcpy(pkey->name, key, 16);
-    ngx_memcpy(pkey->aes_key, key + 16, 16);
-    ngx_memcpy(pkey->hmac_key, key + 32, 16);
+    if (ngx_strlen(key) == 48){
```

Comment:
@fbomlisboa BTW, it is also possible to include null characters (`\0`) in the key data itself. It's completely valid since it's an arbitrary binary string. So this is not just suboptimal but also a bug IMHO.